### ISSUE TITLE
Module directory checksum does not match sum.golang.org

### ISSUE THREAD

**Issue by stefanello57** [vikyd/go-checksum#2]
According to https://sum.golang.org/lookup/github.com/git-chglog/git-chglog@v0.10.0 git-chglog in version 0.10.0 has the following hashes:

- go.mod: `h1:WUa9ClOuSj9JSAGvEtsmVew3urFQPCroKmR2xW3gLeI=`
- module directory: `h1:xMlfLYG8n7PawZGZJaL5Duy4b2Z1kvvCpzhGH5+sUYo=`

Now I ran: `go get -u github.com/git-chglog/git-chglog/cmd/git-chglog`

With go-checksum I get the following results:

**go.mod**
- Command: `go-checksum {my-path}/pkg/mod/github.com/git-chglog/git-chglog@v0.10.0/go.mod`
- GoCheckSum: `h1:WUa9ClOuSj9JSAGvEtsmVew3urFQPCroKmR2xW3gLeI=`

**module directory**
- Command: `go-checksum {my-path}/pkg/mod/github.com/git-chglog/git-chglog@v0.10.0 git-chglog/git-chglog@v0.10.0`
- GoCheckSum: `h1:Mq4RljIwShX51EuGsWyK9KasXdaIsqzWQEsoXZQgz3Y=`

So I do have a match for the go.mod checksum but the module directory checksum is different. How can that be?

**Comment by vikyd** [comment]
Good question!

# TLDR

`go-checksum` works as expected. 

The problem is the author of `github.com/git-chglog/git-chglog` has changed the content of `v0.10.0` after this tag comes out. 


# Long: How to reproduce?
two steps.

## 1. get the checksum the same as: https://sum.golang.org/lookup/github.com/git-chglog/git-chglog@v0.10.0

Because go download `git-chglog` from `GOPROXY`(https://proxy.golang.org/github.com/git-chglog/git-chglog/@v/v0.11.0.zip)
```sh
# cd anywhere
mkdir testchecksum

cd testchecksum

go mod init example.com/abc

go get -x github.com/git-chglog/git-chglog@v0.10.0

cat go.sum | grep git-chglog
```
should see：
```
github.com/git-chglog/git-chglog v0.10.0 h1:xMlfLYG8n7PawZGZJaL5Duy4b2Z1kvvCpzhGH5+sUYo=
github.com/git-chglog/git-chglog v0.10.0/go.mod h1:WUa9ClOuSj9JSAGvEtsmVew3urFQPCroKmR2xW3gLeI=
```
show GOPROXY：
```sh
go env | grep GOPROXY
# should see:
# GOPROXY="https://proxy.golang.org,direct"
```


## 2. get a different checksum
```sh
export GOPROXY=direct

mkdir testchecksum2

cd testchecksum2

go mod init example.com/abc

go get -x github.com/git-chglog/git-chglog@v0.10.0
```

this time, should see some checksum error:
```
go get: github.com/git-chglog/git-chglog@v0.10.0: verifying module: checksum mismatch
        downloaded: h1:/6Sl1zS8arHDvkGiLbAOP6Xu/8XY87wgKhO+gJZkcDk=
        sum.golang.org: h1:xMlfLYG8n7PawZGZJaL5Duy4b2Z1kvvCpzhGH5+sUYo=

SECURITY ERROR
This download does NOT match the one reported by the checksum server.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```

show GOPROXY：
```sh
go env | grep GOPROXY
# should see:
# GOPROXY="direct"
```


## when did the author modify the tag content？
https://proxy.golang.org/github.com/git-chglog/git-chglog/@v/v0.10.0.info time: [2021-01-16T06:04:55Z](https://proxy.golang.org/github.com/git-chglog/git-chglog/@v/v0.10.0.info)

tag [v0.10.0](https://github.com/git-chglog/git-chglog/releases/tag/v0.10.0) corresponding commit time: [2021-01-16T07:05:18Z](https://github.com/git-chglog/git-chglog/commit/73994ac8f46cc42edad5ed6a5d56de0d31f826d9) 

Golang encourages not to modify anything of released versions.

## digression
In question desc: 

`go-checksum {my-path}/pkg/mod/github.com/git-chglog/git-chglog@v0.10.0 git-chglog/git-chglog@v0.10.0` 

should be:

`go-checksum {my-path}/pkg/mod/github.com/git-chglog/git-chglog@v0.10.0 github.com/git-chglog/git-chglog@v0.10.0`
